CI: tests failing on 3.7 builds

clearly something changed for dateutil

```
pytest -m "not slow and not network" -n auto --dist=loadfile -s --strict-markers --durations=30 --junitxml=test-data.xml pandas
ImportError while loading conftest '/Users/runner/work/1/s/pandas/conftest.py'.
pandas/__init__.py:51: in <module>
    from pandas.core.api import (
pandas/core/api.py:24: in <module>
    from pandas.core.algorithms import (
pandas/core/algorithms.py:39: in <module>
    from pandas.core.dtypes.cast import (
pandas/core/dtypes/cast.py:29: in <module>
    from dateutil.parser import ParserError
E   ImportError: cannot import name 'ParserError' from 'dateutil.parser' (/Users/runner/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/dateutil/parser/__init__.py)
```

--- a/pandas/core/dtypes/cast.py
+++ b/pandas/core/dtypes/cast.py
@@ -26,7 +26,6 @@ from typing import (
 )
 import warnings
 
-from dateutil.parser import ParserError
 import numpy as np
 
 from pandas._libs import (
@@ -1588,19 +1587,10 @@ def maybe_cast_to_datetime(
                             value = to_timedelta(value, errors="raise")._values
                     except OutOfBoundsDatetime:
                         raise
-                    except ParserError:
-                        # Note: ParserError subclasses ValueError
-                        # str that we can't parse to datetime
+                    except ValueError:
+                        # TODO(GH#40048): only catch dateutil's ParserError
+                        #  once we can reliably import it in all supported versions
                         pass
-                    except ValueError as err:
-                        if "mixed datetimes and integers in passed array" in str(err):
-                            # array_to_datetime does not allow this;
-                            # when called from _try_cast, this will be followed
-                            #  by a call to construct_1d_ndarray_preserving_na
-                            #  which will convert these
-                            pass
-                        else:
-                            raise
 
         # coerce datetimelike to object
         elif is_datetime64_dtype(

--- a/pandas/tests/frame/test_constructors.py
+++ b/pandas/tests/frame/test_constructors.py
@@ -89,8 +89,6 @@ class TestDataFrameConstructors:
             arr = arr.reshape(1, 1)
 
         msg = "Could not convert object to NumPy timedelta"
-        if frame_or_series is Series:
-            msg = "Invalid type for timedelta scalar: <class 'numpy.datetime64'>"
         with pytest.raises(ValueError, match=msg):
             frame_or_series(arr, dtype="m8[ns]")
 

--- a/pandas/tests/series/test_constructors.py
+++ b/pandas/tests/series/test_constructors.py
@@ -1332,7 +1332,7 @@ class TestSeriesConstructors:
             td.astype("int32")
 
         # this is an invalid casting
-        msg = "Could not convert 'foo' to NumPy timedelta"
+        msg = "Could not convert object to NumPy timedelta"
         with pytest.raises(ValueError, match=msg):
             Series([timedelta(days=1), "foo"], dtype="m8[ns]")
 

